mozilla/blurts-server
# Fix scan result cards expanding/collapsing in sync

# References: 
Jira: MNTOR-2504 (this was about cards not opening in the first place. I can't reproduce that myself, but I'm fairly sure that's the same root cause), MNTOR-2463, MNTOR-2446.
Figma: N/A




# Description

The problem was that getLatestOnerepScanResults would return the result of a joined query of onerep_scan_results with onerep_scans, which would mean the `id`s of the former would get overridden by those of the latter, and thus multiple scan results all appearing to have the same ID.

I applied a couple of redundant fixes:
- `getLatestOnerepScanResults` now only returns data from the `onerep_scan_results` column.
- The dashboard now uses the `onerep_scan_result_id to identify scan results.
- Tests now verify that there is actually only a single card expanded.
- Tests now cover the situation where the dashboard shows multiple unresolved scan results.
- (Not committed because it's no longer representative:) I updated the mock data to give the multiple unresolved scan results the same ID, and verified that that caused the test to fail without the changes to `exposureCardKey`, and to pass _with_ those changes.

# How to test

Open the new dashboard with multiple scan results visible, and open and collapse different cards. There should always be at most one card open.

# Checklist (Definition of Done)
